### ISSUE TITLE
RAVQA-v2 code

### ISSUE THREAD

**Issue by yjwang97** [LinWeizheDragon/Retrieval-Augmented-Visual-Question-Answering#24]
Hello, thank you for your excellent work! May I ask when the code for "Fine-grained Late-interaction Multi-modal Retrieval for Retrieval Augmented Visual Question Answering" will be released?

**Comment by LinWeizheDragon** [comment]
Hi, thank you for your interest in our work.
We will try to release the code as soon as possible, but the situation is that I will be attending conferences this month and will need to close up another project soon. I will try my best to sort out the code and release them. Currently, I think I can finish testing and reproducing results before the end of Jan.

**Comment by yjwang97** [comment]
OK，Thank you for your reply.

**Comment by ningding-o** [comment]
Expect the release of code for FLMR in NIPS23 !!~

**Comment by yjwang97** [comment]
Hello, I've recently been attempting knowledge retrieval using Colbertv2. When I incorporated text-based visual information (similar to the DPR data used in RAVQA), the R@5 was only 82.7. I noticed that the FLMR paper achieved 85.99. I was wondering if the positive and negative examples used in FLMR are the same as those in RAVQA?

**Comment by LinWeizheDragon** [comment]
Hi, I think the training data remains the same in my memory. However, since the RAVQAv1 was released a long time ago, I am not entirely sure whether there are tiny changes in the data. However, I can confirm that the positive samples come from the same annotation file (the GS annotations). And the negative samples come from the training corpus (excluding the test corpus which may leak additional test info).

I would suggest that you try to follow the hyperparameters presented in the paper. As a practical matter, the number of negative examples used in training affects the final performance significantly.
For example, batch size 30 with 1 in-batch negatives achieves a good performance. To achieve similar performance with a smaller batch size, you may need 8 batch size with 4 in-batch negatives.

**Comment by LinWeizheDragon** [comment]
> Expect the release of code for FLMR in NIPS23 !!~

Thanks for your kind words. I will try to release the code as soon as possible. Thanks for your understanding - I have been overwhelmed these days.

**Comment by yjwang97** [comment]
> Hi, I think the training data remains the same in my memory. However, since the RAVQAv1 was released a long time ago, I am not entirely sure whether there are tiny changes in the data. However, I can confirm that the positive samples come from the same annotation file (the GS annotations). And the negative samples come from the training corpus (excluding the test corpus which may leak additional test info).
> 
> I would suggest that you try to follow the hyperparameters presented in the paper. As a practical matter, the number of negative examples used in training affects the final performance significantly. For example, batch size 30 with 1 in-batch negatives achieves a good performance. To achieve similar performance with a smaller batch size, you may need 8 batch size with 4 in-batch negatives.

Thank you very much, now it has normalized the results
